### PR TITLE
add GOPROXY env when fails and retry

### PR DESCRIPTION
**Pull request by Claych**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: scripts/dockerfiles/Dockerfile.plugins [hunk 1]
```diff
@@ -52,5 +52,6 @@ RUN git clone $CLOUDWATCH_PLUGIN_CLONE_URL /cloudwatch
 WORKDIR /cloudwatch
 RUN if [ -n "$CLOUDWATCH_PLUGIN_BRANCH" ];then git fetch --all && git checkout $CLOUDWATCH_PLUGIN_BRANCH && git remote -v;fi
 RUN if [ -z "$CLOUDWATCH_PLUGIN_BRANCH" ];then git fetch --all --tags && git checkout tags/$CLOUDWATCH_PLUGIN_TAG -b $CLOUDWATCH_PLUGIN_TAG && git describe --tags;fi
-RUN go mod download
+RUN go mod download || ( go env -w GOPROXY=direct && go mod download )
+# RUN go mod download || go env -w GOPROXY=direct && echo "set proxy" && go mod download && echo "set proxy"
```
Comment (PettitWesley) [inline scripts/dockerfiles/Dockerfile.plugins:56]: remove echo statements and comment from testing

File: scripts/dockerfiles/Dockerfile.plugins-windows [hunk 2]
```diff
@@ -53,7 +52,7 @@ RUN git clone $CLOUDWATCH_PLUGIN_CLONE_URL /cloudwatch
 WORKDIR /cloudwatch
 RUN if [ -n "$CLOUDWATCH_PLUGIN_BRANCH" ];then git fetch --all && git checkout $CLOUDWATCH_PLUGIN_BRANCH && git remote -v;fi
 RUN if [ -z "$CLOUDWATCH_PLUGIN_BRANCH" ];then git fetch --all --tags && git checkout tags/$CLOUDWATCH_PLUGIN_TAG -b $CLOUDWATCH_PLUGIN_TAG && git describe --tags;fi
-RUN go mod download
+RUN go mod download || ( go env -w GOPROXY=direct && go mod download )
```
Comment (PettitWesley) [inline scripts/dockerfiles/Dockerfile.plugins-windows:55]: why parenthesis around the retry? Is this a bash thing?

Comment (Claych) [reply scripts/dockerfiles/Dockerfile.plugins-windows:55]: without parenthesis, we will always run `go mod download` again in logic. Here are the test case results
Screenshot 2023-03-27 at 2 54 45 PM

Comment (Claych) [reply scripts/dockerfiles/Dockerfile.plugins-windows:55]: Screenshot 2023-03-27 at 3 06 04 PM
5ff8fe0568a4.png">

Comment (Claych) [reply scripts/dockerfiles/Dockerfile.plugins-windows:55]: Screenshot 2023-03-27 at 3 06 04 PM

**Review by PettitWesley — APPROVED** [review]

File: scripts/dockerfiles/Dockerfile.build-init [hunk 3]
```diff
@@ -15,11 +15,13 @@ ENV PATH ${PATH}:/home/.gimme/versions/gostable.linux.arm64/bin:/home/.gimme/ver
...
 ENV GO111MODULE on
-RUN go env -w GOPROXY=direct
 
 # Build init process for Fluent Bit
 COPY /init/fluent_bit_init_process.go /
 COPY /go.mod /
 COPY /go.sum /
-RUN go mod tidy \
-    && go build fluent_bit_init_process.go
+RUN ( go mod tidy \
```
Comment (PettitWesley) [inline scripts/dockerfiles/Dockerfile.build-init:23]: nit: the go mod tidy could run on one line/RUN statement and then the build next, to make this shorter

**Review by PettitWesley — APPROVED** [review]

File: scripts/dockerfiles/Dockerfile.build-init (hunk 3 above)
Comment (Claych) [reply scripts/dockerfiles/Dockerfile.build-init:23]: do you mean go mod tidy doesn't reply on GOPROXY=direct?

Comment (Claych) [reply scripts/dockerfiles/Dockerfile.build-init:23]: will test on the pipeline whether go mod tidy effected by GOPROXY=direct, if so, will let it has its own line

File: scripts/dockerfiles/Dockerfile.build-init [hunk 4]
```diff
@@ -15,11 +15,11 @@ ENV PATH ${PATH}:/home/.gimme/versions/gostable.linux.arm64/bin:/home/.gimme/ver
...
 
 # Build init process for Fluent Bit
 COPY /init/fluent_bit_init_process.go /
 COPY /go.mod /
 COPY /go.sum /
-RUN go mod tidy \
-    && go build fluent_bit_init_process.go
+RUN go mod tidy || ( go env -w GOPROXY=direct && go mod tidy )
+RUN go build fluent_bit_init_process.go \
+    ( go env -w GOPROXY=direct && go build fluent_bit_init_process.go )
```
Comment (PettitWesley) [inline scripts/dockerfiles/Dockerfile.build-init:25]: No || on this one??

**Pull request merged**